#Анализ пользовательского взаимодействия с карточками статей.

## Шаг 1. Загрузим данные и изучим общую информацию.

Загрузим данные и изучим общую информацию о датасете. 

In [ ]:
# импортируем библиотеки
import pandas as pd
from sqlalchemy import create_engine

In [ ]:
db_config = {'user': 'praktikum_student', # имя пользователя
            'pwd': 'Sdf4$2;d-d30pp', # пароль
            'host': 'rc1b-wcoijxj3yxfsf3fs.mdb.yandexcloud.net',
            'port': 6432, # порт подключения
            'db': 'data-analyst-zen-project-db'} # название базы данных

connection_string = 'postgresql://{}:{}@{}:{}/{}'.format(db_config['user'],
                                                db_config['pwd'],
                                                db_config['host'],
                                                db_config['port'],
                                                db_config['db'])

engine = create_engine(connection_string) 

In [ ]:
query = '''SELECT *
           FROM dash_visits
'''

In [ ]:
df = pd.io.sql.read_sql(query, con = engine)
df.head()

In [ ]:
#посмотрим датасет

df.info()
display(df.head(10));

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 30745 entries, 0 to 30744
Data columns (total 6 columns):
 #   Column        Non-Null Count  Dtype 
---  ------        --------------  ----- 
 0   record_id     30745 non-null  int64 
 1   item_topic    30745 non-null  object
 2   source_topic  30745 non-null  object
 3   age_segment   30745 non-null  object
 4   dt            30745 non-null  object
 5   visits        30745 non-null  int64 
dtypes: int64(2), object(4)
memory usage: 1.4+ MB


record_id item_topic source_topic age_segment                   dt  visits
0    1040597     Деньги         Авто       18-25  2019-09-24 18:32:00       3
1    1040598     Деньги         Авто       18-25  2019-09-24 18:35:00       1
2    1040599     Деньги         Авто       18-25  2019-09-24 18:54:00       4
3    1040600     Деньги         Авто       18-25  2019-09-24 18:55:00      17
4    1040601     Деньги         Авто       18-25  2019-09-24 18:56:00      27
5    1040602     Деньги         Авто       18-25  2019-09-24 18:57:00      22
6    1040603     Деньги         Авто       18-25  2019-09-24 18:58:00      24
7    1040604     Деньги         Авто       18-25  2019-09-24 18:59:00      20
8    1040605     Деньги         Авто       18-25  2019-09-24 19:00:00       5
9    1040606     Деньги         Авто       26-30  2019-09-24 18:29:00       1

**Выводы:** 

Общая информация о датасете:

У нас 6 столбцов с данными. Максимальное количество строк 30745.

Большинство столбцов имеют тип: object и float64.

# Шаг 2. Выполним предобработку данных.
## Изучим, есть ли дубликаты в данных. Поищим пропуски: встречаются ли они, в каких столбцах? Решим обработать их или оставить как есть.

In [ ]:
#Проверяем наличие явных дубликатов
print('Количество дубликатов:', df.duplicated().sum())

Количество дубликатов: 0


In [ ]:
#Проверяем наличие пропусков
pd.DataFrame(round(df.isna().sum(),1)).style.background_gradient('coolwarm')

**Вывод**

Явных дубликатов нет.
Пропущенных значения в столбцах нет. 
Удалять или заменять ничего не надо. Аномалий нет.

## Выполним экспорт данных в файл.

In [ ]:
df.to_csv('dash_visits.csv', index = False)

## Дашборд и презентация

Ссылка на дашборд - https://public.tableau.com/app/profile/sergei.iurinov/viz/Dash_Visits_16829607688820/Dashboard1?publish=yes

Ссылка на презентацию - https://disk.yandex.ru/d/uVAinXMH2aRXVg